src/EventStore.Core/TransactionLog/Chunks/TFChunk/TFChunk.cs

```diff
@@ -1155,7 +1194,7 @@ private enum CacheStatus {
 
 			// if we get here, then we reserved TFChunk for sure so no one should dispose of chunk file
 			// until client returns dedicated reader
-			return new TFChunkBulkReader(this, GetSequentialReaderFileStream());
+			return new TFChunkBulkReader(this, GetSequentialReaderFileStream(), isMemory: false);
```

Comment:
yeah we wondered about doing something like that, but since (as you say) GetSequentialReaderStream always incremented the filestreamcount even for inmem, i just decided to look at the memdb case as operating at a higher level of abstraction and tried to interfere with it as little as possible here

Reply:
in the medium to long term i wonder if it would be better to somehow separate the memdb implementation (if we need to keep it at all) it's (sometimes) a bit confusing to have it interleaved in tfchunk

Reply:
> yeah we wondered about doing something like that, but since (as you say) GetSequentialReaderStream always incremented the filestreamcount even for inmem, i just decided to look at the memdb case as operating at a higher level of abstraction and tried to interfere with it as little as possible here

ok sounds good :+1:

Reply:
> in the medium to long term i wonder if it would be better to somehow separate the memdb implementation (if we need to keep it at all) it's (sometimes) a bit confusing to have it interleaved in tfchunk

yup i agree that it's sometimes confusing (and probably error-prone too), maybe abstracting the whole file-based stuff could work